lib/bank_account/account.rb

```diff
@@ -0,0 +1,38 @@
+module BankAccount
+
+  class Account
+
+    def initialize(id, initial_balance)
+      @id = id
+      @checks = 0
+      if initial_balance <= 0
+        raise ArgumentError, "Account cannot be 0 or less"
+      else
+        @balance = initial_balance
+      end
+
+    end
+
+    def withdraw(amount)
+      if @balance - amount <= 0
+        puts "There is not enough money to withdraw $#{amount}."
+      else
+        @balance -= amount
+      end
+      return @balance
+    end
+
+    def deposit(amount)
+      @balance += amount
+      return @balance
+    end
+
+    def balance
```

Comment:
Since you're using an instance variable called `@balance` you could use an `attr_reader` for this variable and then you don't need a separate method